airbyte-config/config-models/src/main/java/io/airbyte/config/helpers/StateMessageHelper.java

```diff
@@ -53,6 +55,25 @@ public static Optional<StateWrapper> getTypedState(final JsonNode state) {
     }
   }
 
+  /**
+   * Converts a StateWrapper to a State
+   *
+   * LegacyStates are directly serialized into the state. GlobalStates and StreamStates are serialized
+   * as a list of AirbyteStateMessage in the state attribute.
+   *
+   * @param stateWrapper the StateWrapper to convert
+   * @return the Converted State
+   */
+  @SuppressWarnings("UnnecessaryDefault")
+  public static State getState(final StateWrapper stateWrapper) {
+    return switch (stateWrapper.getStateType()) {
+      case LEGACY -> new State().withState(stateWrapper.getLegacyState());
+      case STREAM -> new State().withState(Jsons.jsonNode(stateWrapper.getStateMessages()));
+      case GLOBAL -> new State().withState(Jsons.jsonNode(Collections.singletonList(stateWrapper.getGlobal())));
```

Comment:
Nit:  use can use `List.of()` instead of `Collections.singletonList` to avoid the extra import.